kea-vscode/src/views/pull-request-contents/commits/commits-root-tree-node.ts

```diff
@@ -1,34 +1,92 @@
  */
-export class CommitsRootTreeNode implements IParentTreeNode<CommitTreeNode> {
+export class CommitsRootTreeNode implements IParentTreeNode<CommitTreeNode | LocalCommitTreeNode> {
   #contextValue = "commit";
   #iconPath = new vscode.ThemeIcon("git-commit");
   #repository: IKeaRepository;
+  #ctx: IKeaContext;
+  #pullRequest: PullRequest | undefined;
+  #localCommitsService: LocalCommitsService;
 
   pullId: PullRequestId;
   collapsibleState: CollapsibleState = "collapsed";
 
-  constructor(repository: IKeaRepository, pullId: PullRequestId) {
+  constructor(repository: IKeaRepository, pullId: PullRequestId, ctx: IKeaContext, pullRequest?: PullRequest) {
     this.#repository = repository;
     this.pullId = pullId;
+    this.#ctx = ctx;
+    this.#pullRequest = pullRequest;
+    this.#localCommitsService = new LocalCommitsService(ctx, repository.account.accountKey, repository.repoId);
   }
 
   getTreeItem = (): vscode.TreeItem => {
     const treeItem = new vscode.TreeItem("Commits", getCollapsibleState(this.collapsibleState));
     treeItem.contextValue = this.#contextValue;
     treeItem.iconPath = this.#iconPath;
+
+    // Add description in the background (don't await to avoid blocking the tree)
+    void this.#addBranchStatusDescription(treeItem);
+
     return treeItem;
   };
 
-  getChildren = async (): Promise<CommitTreeNode[]> => {
+  #addBranchStatusDescription = async (treeItem: vscode.TreeItem): Promise<void> => {
+    try {
+      // Try to get the first workspace folder to check local git status
+      const workspaceFolders = vscode.workspace.workspaceFolders;
+      if (!workspaceFolders || workspaceFolders.length === 0) {
+        return;
+      }
+
+      const workspaceFolder = workspaceFolders[0];
+      if (!workspaceFolder) {
+        return;
+      }
+
+      const localGitRepo = await this.#ctx.gitManager.getLocalGitRepository(workspaceFolder);
+      if (localGitRepo instanceof Error) {
+        return;
+      }
+
+      const branchStatus = await localGitRepo.getBranchStatus();
+      if (branchStatus instanceof Error) {
+        return;
+      }
+
+      if (branchStatus.ahead > 0 || branchStatus.behind > 0) {
+        const parts: string[] = [];
+        if (branchStatus.ahead > 0) {
+          parts.push(`↑${branchStatus.ahead}`);
+        }
+        if (branchStatus.behind > 0) {
+          parts.push(`↓${branchStatus.behind}`);
+        }
+        treeItem.description = parts.join(" ");
+      }
+    } catch (error) {
+      Logger.debug("Failed to get branch status for tree item description", error);
+    }
+  };
+
+  getChildren = async (): Promise<Array<CommitTreeNode | LocalCommitTreeNode>> => {
+    // First try to get local commits if we're in a workspace
+    const localCommits = await this.#getLocalCommits();
+    if (localCommits && localCommits.length > 0) {
+      return localCommits;
+    }
+
+    // Fall back to remote commits from API
```

Comment:
Please always get remote commits - we need to be able to tell if we're behind or not, and show the remote-only commits.